app/uk/gov/hmrc/hec/services/FileCreationService.scala

```diff
@@ -37,87 +42,169 @@ trait FileCreationService {
+    LicenceTimeTrading.values.map { values =>
+      val keyValue = licenceTimeTradingEKV(values)
+      EnumFileBody(recordId = keyValue._1, recordDescription = keyValue._2)
+    }.toList
+
+  private def createLicenceTypeEnumFileBody =
+    LicenceType.values.map { values =>
+      val keyValue = licenceTypeEKV(values)
+      EnumFileBody(recordId = keyValue._1, recordDescription = keyValue._2)
+    }.toList
+
+  private def createLicenceValidityPeriodFileBody =
     LicenceValidityPeriod.values.map { values =>
-      val keyValue = enumKeysAndValue(values)
+      val keyValue = licenceValidityPeriodEKV(values)
+      EnumFileBody(recordId = keyValue._1, recordDescription = keyValue._2)
+    }.toList
+
+  private def createCorrectiveActionFileBody =
+    CorrectiveAction.values.map { values =>
+      val keyValue = correctiveActionEKV(values)
       EnumFileBody(recordId = keyValue._1, recordDescription = keyValue._2)
     }.toList
+
+  case class TaxSituationMapping(SA: Option[Char], PAYE: Option[Char], NotChargeable: Option[Char])
+
+  private def taxSituationMapping(taxSituation: TaxSituation) = taxSituation match {
+    case PAYE          => TaxSituationMapping(Some('N'), Some('Y'), None)
+    case SA            => TaxSituationMapping(Some('Y'), Some('N'), None)
+    case SAPAYE        => TaxSituationMapping(Some('Y'), Some('Y'), None)
+    case NotChargeable => TaxSituationMapping(None, None, Some('Y'))
+  }
+
+  case class SAStatusMapping(returnReceived: Option[Char], noticeToFileIssued: Option[Char])
+
+  private def saStatusMapping(saStatus: Option[SAStatus]) = saStatus match {
+    case Some(ReturnFound)        => SAStatusMapping(Some('Y'), None)
+    case Some(NoticeToFileIssued) => SAStatusMapping(Some('N'), Some('Y'))
+    case Some(NoReturnFound)      => SAStatusMapping(Some('N'), Some('N'))
+    case None                     => SAStatusMapping(None, None)
   }
+
+  private def createHecTaxCheckFileBody(hecTaxCheckList: List[HECTaxCheck]): List[HECTaxCheckFileBody] =
+    hecTaxCheckList.map { hecTaxCheck =>
+      hecTaxCheck.taxCheckData match {
+        case i: IndividualHECTaxCheckData =>
+          HECTaxCheckFileBody(
+            ggCredID = Some(i.applicantDetails.ggCredId.value),
+            nino = Some(i.taxDetails.nino.value),
+            firstName = Some(i.applicantDetails.name.firstName),
+            lastName = Some(i.applicantDetails.name.lastName),
+            dob = Some(i.applicantDetails.dateOfBirth.value.format(DATE_FORMATTER)),
+            SAUTR = i.taxDetails.sautr.map(_.value.toInt),
+            licenceType = licenceTypeEKV(i.licenceDetails.licenceType)._1,
+            licenceValidityPeriod = licenceValidityPeriodEKV(i.licenceDetails.licenceValidityPeriod)._1,
+            licenceTimeTrading = licenceTimeTradingEKV(i.licenceDetails.licenceTimeTrading)._1,
+            entityType = 'I',
+            notChargeable = taxSituationMapping(i.taxDetails.taxSituation).NotChargeable,
```

Comment:
we could extract out the value of `taxSituationMapping(i.taxDetails.taxSituation)` into a `val` to avoid calling it three times here. We could do a similar thing for `saStatusMapping(i.taxDetails.saStatusResponse.map(_.status))` as well.